driver/src/main/scala/api/api.scala

```diff
@@ -552,26 +569,66 @@ class MongoDriver(systemOption: Option[ActorSystem] = None) {
    */
   def connection(parsedURI: MongoConnection.ParsedURI): MongoConnection =
     connection(parsedURI, 10, None)
+
+  private case class AddConnection(options: MongoConnectionOptions, mongosystem: ActorRef)
+  private case class CloseWithTimeout(timeout: FiniteDuration)
+
+  private case class SupervisorActor(driver: MongoDriver) extends Actor {
+    import MongoDriver.logger
+
+    def isEmpty = driver.connectionMonitors.isEmpty
+
+    override def receive = {
+      case ac: AddConnection =>
+        val connection = new MongoConnection(driver.system, ac.mongosystem, ac.options)
+        driver.connectionMonitors.put(connection.monitor, connection)
+        val actor = connection.monitor
+        context.watch(actor)
+        sender ! connection
+      case Terminated(actor) =>
+        driver.connectionMonitors.remove(actor)
+      case CloseWithTimeout(timeout) =>
+        if (isEmpty) {
+          context.stop(self)
+        } else {
+          context.become(closing(timeout))
+        }
+      case Close =>
+        if (isEmpty) {
+          context.stop(self)
+        } else {
+          context.become(closing(0.seconds))
+        }
+    }
+
+    def closing(shutdownTimeout: FiniteDuration) : Receive = {
+      case ac: AddConnection =>
+        logger.warn("Refusing to add connection while MongoDriver is closing.")
+      case Terminated(actor) =>
+        driver.connectionMonitors.remove(actor)
+        if (isEmpty) {
+          context.stop(self)
+        }
+      case CloseWithTimeout(timeout) =>
+        logger.warn("CloseWithTimeout ignored, already closing.")
+      case Close =>
+        logger.warn("Close ignored, already closing.")
+    }
+
+    override def postStop {
+      driver.system.shutdown()
+    }
+  }
 }
 
 object MongoDriver {
   private val logger = LazyLogger("reactivemongo.api.MongoDriver")
 
-  /** Default ActorSystem used in the default MongoDriver constructor. */
-  private def defaultSystem = {
-    import com.typesafe.config.ConfigFactory
-    val config = ConfigFactory.load()
-    ActorSystem("reactivemongo", config.getConfig("mongo-async-driver"))
-  }
-
   /** Creates a new MongoDriver with a new ActorSystem. */
   def apply() = new MongoDriver
 
-  /**
-   * Creates a new MongoDriver with specified ActorSystem.
-   *
-   * @param system An ActorSystem for ReactiveMongo to use.
-   */
-  def apply(system: ActorSystem) = new MongoDriver(system)
+  def apply(config: Config) = new MongoDriver(Some(config))
 
-}
\ No newline at end of file
+  private[api] val _counter = new AtomicLong(0)
+  def nextCounter : Long = _counter.incrementAndGet()
```

Comment:
Is there any reason to expose this as a public API?  Unless it is required to do so, I think we should reduce its visibility to the package.